---
title: "FigsForPaper"
author: "Sophia Frantz"
date: "10/28/2019"
output: html_document
---

files and packages:
```{r  message=FALSE}
#Differential expression files: 
batch1_Geno_noOutlier = read.csv(file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/resultsGenoWithinBatch1_noOutlier", header = TRUE)
batch2_Geno = read.csv(file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/BatchAndOutliers/resultsGenoWithinBatch2", header=TRUE )
resultsRemoveOutlier = read.csv(file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/BatchAndOutliers/resultsTrtControlGeno_RemoveOutlier", header = TRUE, row.names=1) 

#counts files for permanova: 
countDat_batch1 = read.csv(file ="/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/countDat_batch1", header = TRUE, row.names = 1 )
countDat_batch2 = read.csv(file ="/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/countDat_batch2", header = TRUE, row.names = 1 )
#metadata 
metadata_batch1 = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/batch1", header=TRUE, row.names = 1) 
metadata_batch2 = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/batch2", header=TRUE, row.names = 1)  
  
#genes list for Venn Diagrams 
batch1_genes = batch1_Geno_noOutlier[which(batch1_Geno_noOutlier$padj < 0.05),]
batch1_genes_down = batch1_genes[which(batch1_genes$log2FoldChange < 0),]
batch1_genes_up = batch1_genes[which(batch1_genes$log2FoldChange > 0),]
batch2_genes = batch2_Geno[which(batch2_Geno$padj < 0.05),]
batch2_genes_down = batch2_genes[which(batch2_genes$log2FoldChange < 0),]
batch2_genes_up = batch2_genes[which(batch2_genes$log2FoldChange > 0),]
together_genes = resultsRemoveOutlier[which(resultsRemoveOutlier$padj < 0.05),]
together_genes_down = together_genes[which(together_genes$log2FoldChange < 0),]
together_genes_up = together_genes[which(together_genes$log2FoldChange > 0),] 
```


```{r}
#install.packages("vegan")  #the adonis() function will do the permutation analysis
library(vegan)
library(tibble)
library(dplyr)
library(ggplot2)
library(knitr)
library(kableExtra)
```

#Data Quality: Batch 1 and 2 
-percent reads mapped to genome 
-number of DE genes - both methods, both batches. (maybe put this in the 3rd section of results?)


Comparison of Genes identified in Batch 1 vs. Batch 2:
```{r message= FALSE, echo=FALSE}

library(VennDiagram)
vennD = venn.diagram(x = list(batch1_genes$X, batch2_genes$X), category.names = c("batch1", "batch2"), filename = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/Plots_June/batches_design2_comparison.png")
library(imager)
#myimg <- load.image("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/Plots_June/batches_design2_comparison.png" )
#plot(myimg, axes = FALSE)

```

# NMDS Plot 
```{r  message = FALSE, echo=FALSE}
library(vegan)
library(MASS)
countDat3 = read.csv("/Users/sophia/Dropbox/Round2/DESeq_dataframes/countDat_for_sva", header = TRUE, row.names = 1)
#remove the outlier sample! 
countDat3= subset(countDat3, select = -c(RAL73w_plus_1_S13_L004_R1_001.fastq.gzAligned.sortedByCoord.out.bam) )
metadata = read.csv("/Users/sophia/Dropbox/Round2/DESeq_dataframes/metadata_withBatch_withoutOutlier.csv", header=TRUE, row.names = 1) 
countDat3 = t(countDat3)
#png(filename = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/NMDS.png")
par(mgp =c(2.5,1,0))
genotype1=as.character(metadata$genotype)

vare.mds = metaMDS(countDat3, k=3)
```

```{r}
#pdf("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/NMDSLegend")
fig = ordiplot(vare.mds$points, main = "Samples in Transcript Space", ylab="NMDS Dimension 2", xlab="NMDS Dimension 1", font.lab=2, font.axis=2, cex.axis=.7,type="none", cex.main=1)

ordiellipse(vare.mds$points,groups=genotype1, label = TRUE, lwd=2, show.groups = genotype1[1:8], col="cyan3", draw="lines")

ordiellipse(vare.mds$points,groups=genotype1, label = TRUE, lwd=2, show.groups = genotype1[9:15], col="orange3", draw="lines")

ordiellipse(vare.mds$points,groups=genotype1, label = TRUE, lwd=2, show.groups = genotype1[16:23], col="pink2", draw="lines")

ordiellipse(vare.mds$points,groups=genotype1, label = TRUE, lwd=2, show.groups = genotype1[24:31], col="yellow3", draw="lines")

points(fig, "sites", pch=c(rep(1,4),rep(19,4),rep(0,4), rep(15,3),rep(2,4),rep(17,4),rep(5,4),rep(18,4)), col=c(rep("cyan3",8),rep("orange3",7),rep("pink2",8),rep("yellow3",8)), cex=1.5)

legend("bottomleft", legend = c("RAL73","RAL783", "RAL306", "RAL853"), pch = c(0,2,1,5),
col = c("orange3","pink2","cyan3","yellow3"))
dev.off()

#outline = uninfected, filled shapes = infected 
```


## perMANOVA results table 
```{r}
perm_results = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/forTable2", sep = "\t", header = TRUE)
names(perm_results) = c( "", "F-value", "R2", "p-value", "F-value", "R2", "p-value", "F-value", "R2", "p-value")

library(htmlTable)
htmlTable(perm_results,
          cgroup = c("factor", "batch 1", "batch2", "combined"),
          n.cgroup = c(1,3,3,3),
          rnames = FALSE, )
```

to do: include df, SS, MS, remove "combined" columns 

# Tables of GO Terms 
## Batch 1 
```{r}
batch1GO = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/batch1_combinedUpDown_NestedGO", sep = "\t", header = FALSE)
#batch1GO_UpSubset = batch1GO_Up[1:35,]
batch1GO$involved = c("yes","no","yes","yes","yes","yes","no ","yes","no","no","no","yes","no","yes","yes","no","no","yes","yes","yes","no","no","no","yes","no","yes","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","yes","no","yes","no","no","no","no","no","yes","no","yes","no","no","yes","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","yes","no","no","no","no","no","no","no","no","no","no","no")
names(batch1GO) = c("GO biological process", "number of genes in genome", "number differentially expressed", "expected", "over/underrepresented", "enrichment", "raw p-value", "FDR","potentially involved in meiosis?") #add column: involved in meiosis? Also add the gene names, maybe upregulated genes and downregulated genes in a separate file.
kable(batch1GO[,c(1,2,3,6,8)], digits = 3) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"))  %>%
  save_kable("Rplot_Batch1FinalGO.jpg")

#Q: ask people (amanda) to look over the potentially involved in meiosis column. 
```

## Batch 2 
- There is no FDR for Batch2 because there are no significant GO terms after correcting for FDR in Batch 2. So I am displaying the GO Terms with not corrected p-value of less than 0.05. 

- ** Change FDR to 0.1 ?? What is the FDR here?? 

```{r}
batch2GO = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/batch2_combinedUpDown_NestedGO", sep = "\t", header = FALSE)
#batch1GO_UpSubset = batch1GO_Up[1:35,]
batch2GO$involved = c("no", "yes", "no", "no", "no", "yes", "no", "yes", "no", "no", "yes","no","no", "no", "yes", "no", "yes", "no", "yes", "no", "yes", "no", "no", "no", "no", "yes", "no", "yes", "no", "no", "yes", "yes", "no", "no", "no", "yes", "no", "no", "no", "no", "no", "no", "no", "yes", "yes","no", "no", "no", "no", "no", "no", "yes", "yes", "no","no")
names(batch2GO) = c("GO biological process", "number of genes in genome", "number differentially expressed", "expected", "over/underrepresented", "enrichment", "raw p-value", "potentially involved in meiosis?") 
#add column: involved in meiosis? Also add the gene names, maybe upregulated genes and downregulated genes in a separate file.
# No FDR for batch2 
kable(batch2GO[,c(1,2,3,6,7)], digits = 3) %>%
  kable_styling(bootstrap_options = c("striped", "condensed")) %>%
  save_kable("Rplot_Batch2FinalGO_Table.jpg")


#Q: ask people (amanda) to look over the potentially involved in meiosis column. 
```
 To do: Get rid of a lot of this information... I have notes on this somewhere. Include the genes in the table. 
        Get rid of the "involved in meiosis?" column. 
        
# Tables ! 
- tables of up and down- regulated genes for each batch. Tables of GO terms. (need to do KEGG terms too.)

```{r}
library(knitr)
library(kableExtra)

# add a column to depict whether a gene is UP or DOWN-regulated, in case I want to color the boxes: 
newColumn = rep(NA, nrow(batch1_genes))
for (x in 1:nrow(batch1_genes)) {
  y = ifelse(batch1_genes$log2FoldChange[x] < 0, "-", "+" )
  newColumn[x] = y 
}
batch1_genes$Direction = newColumn

test = read.csv(file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/testBatch1_1")

test = sort(test$submitted_id)

write.csv(test, file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/testBatch1_1")

batch1_genes$FoldChange = 2^batch1_genes$log2FoldChange
write.csv(batch1_genes, file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/batch1_df_DE")

batch1_final = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/batch1_df_DE1", header = FALSE)
join = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/batch1_geneSymbols")
batch1_final$geneDescription = join$current_symbol 
batch1_final$V3 = as.numeric(as.character(batch1_final$V3))
batch1_final$V11 = as.numeric(as.character(batch1_final$V11))
batch1_final$V8 = as.numeric(as.character(batch1_final$V8))
#Modify this: 
kable(batch1_final[,c(2,12,3,10,11,8)], digits = 3, col.names = c("FlyBase ID","gene Symbol", "base mean", "direction", "fold change", "adj p-value")) %>%
  kable_styling(bootstrap_options = c("striped", "condensed"))  %>%
  
 # row_spec(which(df$cyl >6), bold = T, color = "white", background = "red")
  save_kable("kable_DEgeneTable.pdf")

# do the same for batch 1 
# fix "direction" later , sort by highest fold change ?? 

# already have GO term tables (for most nested) - in Figs for paper 
```

# Common DE genes between batches 

The genes in common betwen the two batches provide promising candidates for follow-up studies. Of the 20 genes, X have known functions that may be related to meiosis, including chromatin organization, blah, blah. 
Also - immune system response to gram negative bacteria, RNA processing, and metabolism. 
 
- To do: Clean up this table. Just show the gene symbol, full name,  GO terms, and fold change for both Batch1 and Batch2  
- Put the Venn Diagram underneath this table!

## Genes in Common Table:
```{r}
library(dplyr)
library(kableExtra)
table1 = read.csv("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/GeneLists_dir/TableForFigure2", sep = "\t", header = TRUE)
names(table1) = c("Gene Name", "Gene Snapshot (FlyBase)", "GO Terms biological process (Panther)", "potentially involved in meiosis?")
kable(table1[,c(1,3,4)], digits = 3) %>%
  kable_styling(bootstrap_options = c("striped", "condensed")) %>%
  save_kable("Rplot_CommonBatches.png")
```


# Add genes in a separate column 

#  Runs test for Randomness 
-get the position of each gene 
-sort unique this list
-match up the position data with the expression data, sort by position 
  -may be tricky to do genome-wide, do it by chromosome-level. 
-convert the expression data into 0 or 1 --> 0 means not differentially expressed, 1 means diff. expressed. 
-run the randomness test for each chromosome. 

```{r}
# the dataset. Need to parse this file to get 1 position for each gene. Save the name of the gene and the start position. 

```

Location: /Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/RunsTest_GenePositions

columns information: 
bin	name,	chrom	strand,	txStart,	txEnd,	cdsStart	cdsEnd	exonCount	exonStarts	exonEnds	score	name2	cdsStartStat	cdsEndStat	exonFrames

In bash: 
cat 2L_positions | cut -f5,13 | sort -u -k1 -n > 2L_positionsSort
cat 2L_positionsSort | sort -u -k2 > 2L_positionsSorted 
cat 2L_positionsSorted | sort -k1 -n > 2L_positionsSort 

script --> StartPosition_Script 

There is a separate file for each chromosome. Options: 1. add a "chrom" column to keeo track of chromosomes, and concatenate all into one file, or   2. run analyses separately for each chromosome

Then, I need to deal with the dataframe of output from the differential expression analysis. I need the gene name and their expression data. Then I need to add another column. The genes are currently in FlyBase format, so I need to convert them to Ensembl, the same format of the StartPosition dataframe. Or vice-versa. 

Use this gene expression file: 

batch1_Geno_noOutlier =  /Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/resultsGenoWithinBatch1_noOutlier

batch2_Geno = /Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/BatchAndOutliers/resultsGenoWithinBatch2

Modify this file in R? Would that be easier? I need to make a column of + and -. If it is diff regulated, use + . Diff regulated is the same definition: padj <= 0.05 and log2foldChange > 0. 

```{r}
batch1_Geno_noOutlier = read.csv(file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/resultsGenoWithinBatch1_noOutlier", header = TRUE)

newColumn = rep(NA, nrow(batch1_Geno_noOutlier))

for (i in 1:nrow(batch1_Geno_noOutlier)) {
  x = ifelse(abs(batch1_Geno_noOutlier$log2FoldChange[i]) > 0 & batch1_Geno_noOutlier$padj[i] <= 0.05, "+", "-")
  newColumn[i] = x
}

geneNames = batch1_Geno_noOutlier$geneName
geneNames_Runs_Batch1 = data.frame(geneNames = batch1_Geno_noOutlier$geneName,newColumn)
write.csv(geneNames_Runs_Batch1, file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/GenePositions/geneNames_Runs_Batch1")

batch1_Geno_noOutlier$Runs = newColumn

```


-^ And done! There are many "NA" in the file, instead of + or -. This is because either there is  no value for padj or because there is no information for this gene besides the gene name. I think the right thing to do is change instances of NA to "-". This is because we need to have data of + or - for every position in the genome. I can replace NA with - if we can assume that NA means that this gene is not differentially expressed, or stated more conservatively, it was not determined to be differentially expressed in this analysis.  
I am not completely sure though. Maybe it is misleading to code NA as -. 
I will try the Runs test both ways. 

-Ok so the next challenge is converting gene names and then matching up the gene names from the 2 files: gene expression file and the gene position file. I may need to use Python? 

1. Convert gene names: I converted names using FlyBase. The files are in the same /Round2/GenePositions folder, labeled chrom_FB_names. For example, 2L_FB_genes. 

- cat 3L_positionsSort | cut -f2 > 3L_Names 
- Then submit this list to Flybase, save as file- conversion table
- name it 3L_FB_names
- save only the names and not the other information: cat X_FB_names.txt | cut -f2 > X_FB_names 
- remove the other file: rm X_FB_names.txt 

2. match up the gene positions and gene expression data. Maybe sort -u both of them. Then do common and just grab the ones in common. 
  how to deal with the different chromosomes: do 1 at a time. 
  sed 's/"//g'  geneNames_AndRuns_Batch1 > geneNames_AndRuns_Batch1_1
  comm -12 geneNames_AndRuns_Batch1_1  2L_FB_names 

This is not working... a different approach with Grep... 

* You first need a file with both the FBGN and the symbol names. THEN you can match up the gene names with positions, then match up gene names with + - data. 

THIS WORKED: 

for x in $(cut -f2 2L_unsort_posGene); do grep $x 2L_FB_Names_short.txt >> 2L_Symbol_FBgn_grep; done

join -1 1 -2 2 2L_Symbol_FBgn_grep 2L_positionsSort > Joined_2L 

Now joining it with the + and - data is not working though: 
join -1 1  -2 1 -t, geneNames_AndRuns_Batch1_1 Joined_2LTest3

I changed the name of the columns so that the column of gene names would be matching in each file. They both say "geneNames" now. Then I ran thew command. But the output is weird, it is not giving me very many matches there should be much more...

First, grep the lines from the big file with + and - using the file with the gene names and positions:

for x in $(cut -f1 -d"," Joined_2LTest3); do grep $x geneNames_AndRuns_Batch1_1 >> test; done

Then, for some reason it greps the whole line (including + and -) but also just the gene, so select only the lines that contain the + and - information: 

cat test | grep , > test2 

Then do join: 

join -1 1  -2 1 -t, Joined_2LTest3 test2 >> test3Joined 

I don't understand why there is such a reduction when I join these two. The 2 files have ~ 3000 lines, then when I join them they have  1166 lines (test3Joined)
I think I need to sort it first?? That could be why it is messing up. 

--> Ok I think the join function also only retains unique values of the field it is sorting. 

I sort -u both files: 
sort -u -k1 Joined_2LTest3 | wc -l   = 1848
sort -u -k1 test2 | wc -l   = 1764 

I sorted them. 
Then I had to edit to take out some extraneous information on the top and the bottom (head and tail)

Trying to join them:  join -1 1 -2 1 -t , test4 test5 > test6

This worked!    wc -l test6 = 1763 (makes sense given above).

Then I need to put this file back in the order of the order in the genome. I can do this by simply sorting numerically (-n) by the column/field (-k) that contains the start position location. Great!

sort -n -k3 test6  -t"," > test7

join -1 1 -2 2 geneNames_AndRuns_Batch1_2  3R_joinSymbolFB_pos | sed $'s/ /\t/g' | sort -n -u -k4 > 3R_Final 

Done! 
 
 
2R: 
get FBgn from flybase, save file
cut -f1,2 2R_FlybaseFile.txt > 2R_GeneConversions   
sort -k2 -o 2R_positionsSort , sort -k1 -o 2R_GeneConversions 
join -1 2 -2 1 2R_positionsSort 2R_GeneConversions > 2R_FB_position
Next: grep for the genes in the big + - file, then join them. 

# Visualizing changes in transcript space 

## expression across the genome - batch1 up and down-regulated 
```{r echo = FALSE, message = FALSE}
batch1_Geno_noOutlier = read.csv(file = "/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/DESeq_dataframes/resultsGenoWithinBatch1_noOutlier", header = TRUE, row.names=1)
  # YOU HAVE TO make sure you set the rownames as the gene names!!! 
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
txdb <- TxDb.Dmelanogaster.UCSC.dm6.ensGene
dm.genes <- genes(txdb)
# add my data of differential expression information to this genes table. 
mcols(dm.genes) <- batch1_Geno_noOutlier[names(dm.genes), c("log2FoldChange", "stat", "pvalue", "padj")] #make sure results is a dataframe with 6 columns, gene names should be row names, not a column.
# we specify to karyoploteR which genome we are working with (drosophila, 6th annotation):
library(karyoploteR)      # YOU HAVE TO make sure you set the rownames as the gene names!!! 
kp <- plotKaryotype(genome="dm6")
# Label the 10 most differentially expressed genes on the karyotype
#ordered <- dm.genes[order(dm.genes$log2FoldChange, na.last = TRUE),]
kp <- plotKaryotype(genome="dm6")
#kp <- kpPlotMarkers(kp, ordered[1:10], labels = names(ordered[1:10]), text.orientation = "horizontal")
# create a plot with the p-values of all significant genes. First, transform the adjusted p-value into something easy to represent and visualize, such as a minus logarithm transformation, a common trick. Also filter out the genes with NA in the padj column:
filtered.dm.genes <- dm.genes[!is.na(dm.genes$padj)]
log.pval <- -log10(filtered.dm.genes$padj)
mcols(filtered.dm.genes)$log.pval <- log.pval
filtered.dm.genes #there are 9,226 genes in this dataframe now.
#plot the significant genes' log adjusted p-value at their position in the genome on the karyotype:
sign.genes <- filtered.dm.genes[filtered.dm.genes$padj < 0.05,]
kp <- plotKaryotype(genome="dm6")
kpPoints(kp, data=sign.genes, y=sign.genes$log2FoldChange)
#set the ymax 
range(sign.genes$log2FoldChange) # see the range of y-values so we know what to plot on the y-axis (1.3 - 8.0)
```

Chr 2:
```{r echo = FALSE, mesaage = FALSE}
pdf("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/Chr2")
range(sign.genes$log2FoldChange) # the range is about -4 to 8 
#fc.ymax <- ceiling(max(abs(range(sign.genes$log2FoldChange))))
#fc.ymin <- -fc.ymax
fc.ymax =5
fc.ymin=-5
#make the dot represent the adjusted p-value significance - bigger = more significant
#points.top <- 0.8
cex.val <- sqrt(sign.genes$log.pval)/1.5
col.over <- "#FFBD07AA"
col.under <- "#00A6EDAA"
sign.col <- rep(col.over, length(sign.genes))
sign.col[sign.genes$log2FoldChange<0] <- col.under

kp <- plotKaryotype(genome="dm6", chromosomes = c("chr2L", "chr2R"), plot.type = 3)  # Change plot.type?? 
kpPoints(kp, data=sign.genes, y=sign.genes$log2FoldChange, cex = cex.val, ymax=fc.ymax, ymin=fc.ymin, r1=points.top, col=sign.col)
kpAxis(kp, ymax=fc.ymax, ymin=fc.ymin, r1=points.top, cex = 0.5)
kpAddLabels(kp, labels = "log2 fold change", cex = 0.8, srt=90, pos=1, label.margin = 0.045, ymax=fc.ymax, ymin=fc.ymin, r1=points.top)
gene.mean <- start(top.genes) + (end(top.genes) - start(top.genes))/2
#kpSegments(kp, chr=as.character(seqnames(top.genes)), x0=gene.mean, x1=gene.mean, y0=top.genes$log2FoldChange, y1=fc.ymax, ymax=fc.ymax, ymin=fc.ymin, r1=points.top)
#kpPlotMarkers(kp, top.genes, labels = names(top.genes), text.orientation = "horizontal", r0=points.top)

#Data panel 2
kp2 <- kpPlotDensity(kp, data=dm.genes, window.size = 10e4, data.panel = 2, labels = "gene density")
#kpAddLabels(kp2, labels = "gene density", cex = 0.8, srt=90, pos=1, label.margin = 0.04)
dev.off()
```

Chromosome 3: 

```{r echo = FALSE, mesaage = FALSE}
pdf("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/Chr3")
kp <- plotKaryotype(genome="dm6", chromosomes = c("chr3L", "chr3R"), plot.type = 3)  # Change plot.type?? 
kpPoints(kp, data=sign.genes, y=sign.genes$log2FoldChange, cex = cex.val, ymax=fc.ymax, ymin=fc.ymin, r1=points.top, col=sign.col)
kpAxis(kp, ymax=fc.ymax, ymin=fc.ymin, r1=points.top, cex = 0.5)
kpAddLabels(kp, labels = "log2 fold change", cex = 0.8, srt=90, pos=1, label.margin = 0.045, ymax=fc.ymax, ymin=fc.ymin, r1=points.top)
gene.mean <- start(top.genes) + (end(top.genes) - start(top.genes))/2
#kpSegments(kp, chr=as.character(seqnames(top.genes)), x0=gene.mean, x1=gene.mean, y0=top.genes$log2FoldChange, y1=fc.ymax, ymax=fc.ymax, ymin=fc.ymin, r1=points.top)
#kpPlotMarkers(kp, top.genes, labels = names(top.genes), text.orientation = "horizontal", r0=points.top)

#Data panel 2
kp2 <- kpPlotDensity(kp, data=dm.genes, window.size = 10e4, data.panel = 2)
#kpAddLabels(kp2, labels = "gene density", cex = 0.8, srt=90, pos=1, label.margin = 0.04)

# Label the points that are over 4X ? There are only 2. 
dev.off()
```

Chromosome 4 and X:
```{r echo = FALSE, mesaage = FALSE}
pdf("/Users/sophia/Documents/RNAseq_Dmel_Wol/Round2/FinalResults_andFigures_ForPaper/ChrX")
kp <- plotKaryotype(genome="dm6", chromosomes = c("chr4", "chrX"), plot.type = 3)  # Change plot.type?? 
kpPoints(kp, data=sign.genes, y=sign.genes$log2FoldChange, cex = cex.val, ymax=fc.ymax, ymin=fc.ymin, r1=points.top, col=sign.col)
kpAxis(kp, ymax=fc.ymax, ymin=fc.ymin, r1=points.top, cex = 0.5)
kpAddLabels(kp, labels = "log2 fold change", cex = 0.8, srt=90, pos=1, label.margin = 0.045, ymax=fc.ymax, ymin=fc.ymin, r1=points.top)
gene.mean <- start(top.genes) + (end(top.genes) - start(top.genes))/2
#kpSegments(kp, chr=as.character(seqnames(top.genes)), x0=gene.mean, x1=gene.mean, y0=top.genes$log2FoldChange, y1=fc.ymax, ymax=fc.ymax, ymin=fc.ymin, r1=points.top)
#kpPlotMarkers(kp, top.genes, labels = names(top.genes), text.orientation = "horizontal", r0=points.top)

#Data panel 2
kp2 <- kpPlotDensity(kp, data=dm.genes, window.size = 10e4, data.panel = 2)
#kpAddLabels(kp2, labels = "gene density", cex = 0.8, srt=90, pos=1, label.margin = 0.04)
dev.off()
```

To Do: upregulated means more in infected  - be sure to state in the paper and in the figure what the comparison is. 
legend for color, legend for dot? 
distance btw chr 4 and X , or a vertical line btw 4 and X. (I think just more distance. Can just modify this outside of R, in powerpoint.)
If there's not a pattern, do not include gene density - Check the Runs test for randomness first, then if there is no pattern, get rid of the gene density plots. (if there is a pattern...ask.)
abs value log fold change, correlated with gene density, see if there is a correlation. â€“ run's test for randomness 
physical line! 



# KEGG pathways 

## Identification of KEGG terms using GAGE 

```{r message = FALSE, echo = FALSE}
library(pathview)
library(gage)
library(ggplot2)
row.names(batch1_Geno_noOutlier) = batch1_Geno_noOutlier$X
batch1_Geno_noOutlier1 = batch1_Geno_noOutlier
batch1_Geno_noOutlier1$X = NULL
#convert flybase ID to Entrez ID 
id.map.refseq <- id2eg(ids = row.names(batch1_Geno_noOutlier1), org="Dm", category = c("FLYBASE")) 
entrezbatch1_Geno_noOutlier1 <- mol.sum(mol.data = batch1_Geno_noOutlier1, id.map = id.map.refseq, sum.method = "mean")
A_entrezbatch1_Geno_noOutlier1 = as.data.frame(entrezbatch1_Geno_noOutlier1)
library(tidyverse)
deseq2.res <- A_entrezbatch1_Geno_noOutlier1
out.suffix="deseq2"

deseq2.res = rownames_to_column(deseq2.res)
deseq2.res = deseq2.res[,c(1,3)]
deseq2.res = na.omit(deseq2.res)
deseq2.res = remove_rownames(deseq2.res)
deseq2.res= column_to_rownames(deseq2.res, var = 'rowname')
exp.fc = deseq2.res
#need a vector of log2 fold change values

```

This is the main function of the program, gage(). It will determine the KEGG pathway enrichment. 
```{r message = FALSE}
require(gage)
dmel=kegg.gsets(species = "dme", id.type = "entrez", check.new=FALSE)  #id.type can be "kegg" or "entrez". Entrez Gene is the primary KEGG gene ID for many common model organisms, so these two options have the same effect. For non-model organisms, primary KEGG gene IS is not the same as Entrez ID. I converted by flybase IDs to Entrez IDs in the code above.

fc.kegg.p.UpAndDown <- gage(exp.fc, gsets = dmel$kg.sets, ref = NULL, samp = NULL, same.dir=TRUE) #genes as rows and samples as columns 
fc.kegg.p.UpAndDown
#same.dir = TRUE means it will analyze up- and down-regulated separately. 
```


For Batch 2:
```{r message = FALSE, echo = FALSE}
library(pathview)
library(gage)
library(ggplot2)
row.names(batch2_Geno) = batch2_Geno$X
batch2_Geno1 = batch2_Geno
batch2_Geno1$X = NULL
#convert flybase ID to Entrez ID 
id.map.refseq <- id2eg(ids = row.names(batch2_Geno1), org="Dm", category = c("FLYBASE")) 
entrezbatch2_Geno1 <- mol.sum(mol.data = batch2_Geno1, id.map = id.map.refseq, sum.method = "mean")
A_entrezbatch2_Geno1 = as.data.frame(entrezbatch2_Geno1)
library(tidyverse)
deseq2.res <- A_entrezbatch2_Geno1
out.suffix="deseq2"

deseq2.res = rownames_to_column(deseq2.res)
deseq2.res = deseq2.res[,c(1,3)]
deseq2.res = na.omit(deseq2.res)
deseq2.res = remove_rownames(deseq2.res)
deseq2.res= column_to_rownames(deseq2.res, var = 'rowname')
exp.fc = deseq2.res
#need a vector of log2 fold change values

```

This is the main function of the program, gage(). It will determine the KEGG pathway enrichment. 
```{r message = FALSE}
require(gage)
dmel=kegg.gsets(species = "dme", id.type = "entrez", check.new=FALSE)  #id.type can be "kegg" or "entrez". Entrez Gene is the primary KEGG gene ID for many common model organisms, so these two options have the same effect. For non-model organisms, primary KEGG gene IS is not the same as Entrez ID. I converted by flybase IDs to Entrez IDs in the code above.

fc.kegg.p.UpAndDown <- gage(exp.fc, gsets = dmel$kg.sets, ref = NULL, samp = NULL, same.dir=TRUE) #genes as rows and samples as columns 
fc.kegg.p.UpAndDown
#same.dir = TRUE means it will analyze up- and down-regulated separately. 
```


# He et al criteria stuff 

```{r}
nrow(batch1_genes[which(batch1_genes$log2FoldChange > 1),])
nrow(batch1_genes[which(batch1_genes$log2FoldChange < -1),])

nrow(batch2_genes[which(batch2_genes$log2FoldChange < -1),])
nrow(batch2_genes[which(batch2_genes$log2FoldChange > 1),])
```

